packages/harmony/src/components/button/FilterButton/FilterButton.tsx

```diff
@@ -0,0 +1,218 @@
+    }
+    const smallIconStyles: CSSObject = {
+      width: spacing.unit3,
+      height: spacing.unit3
+    }
+
+    const fillContainerStyles: CSSObject = {
+      background: color.secondary.s400,
+      border: `1px solid ${color.secondary.s400}`,
+      '&:hover': {
+        border: `1px solid ${color.secondary.s400}`,
+        transform: 'none'
+      }
+    }
+
+    const activeStyle =
+      variant !== FilterButtonType.FILL_CONTAINER || selection === null
+        ? {
+            border: `1px solid ${color.border.strong}`,
+            background: color.background.surface2
+          }
+        : {}
+
+    // Button Styles
+    const buttonCss: CSSObject = {
+      background: 'transparent',
+      border: `1px solid ${color.border.strong}`,
+      borderRadius: cornerRadius.s,
+      color:
+        variant === FilterButtonType.FILL_CONTAINER && selection !== null
+          ? color.special.white
+          : color.text.default,
+      gap: spacing.xs,
+      fontSize: typography.size.s,
+      fontWeight: typography.weight.medium,
+      lineHeight: typography.lineHeight.s,
+
+      '&:hover': {
+        border: `1px solid ${color.neutral.n800}`,
+        transform: 'none'
+      },
+
+      '&:active': {
+        ...activeStyle,
+        transform: 'none'
+      },
+
+      ...(size === FilterButtonSize.SMALL ? smallStyles : defaultStyles),
+      ...(isOpen ? activeStyle : {}),
+      ...(variant === FilterButtonType.FILL_CONTAINER && selection !== null
+        ? fillContainerStyles
+        : {})
+    }
+
+    const iconCss =
+      size === FilterButtonSize.SMALL ? smallIconStyles : defaultIconStyles
+
+    // Popup Styles
+    const optionCss: CSSObject = {
+      background: 'transparent',
+      border: 'none',
+      color: color.text.default,
+      fontWeight: typography.weight.medium,
+      gap: spacing.s,
+      paddingLeft: spacing.m,
+      paddingRight: spacing.m,
+      paddingTop: spacing.s,
+      paddingBottom: spacing.s,
+      width: '100%',
+      borderRadius: cornerRadius.s,
+      justifyContent: 'flex-start',
+
+      '&:hover': {
+        transform: 'none',
+        backgroundColor: color.secondary.s300,
+        color: color.special.white
+      },
+
+      '&:active': {
+        transform: 'none'
+      }
+    }
+    const optionIconCss: CSSObject = {
+      width: spacing.unit4,
+      height: spacing.unit4
+    }
+
+    const handleButtonClick = useCallback(() => {
+      if (variant === FilterButtonType.FILL_CONTAINER && selection !== null) {
+        setSelection(null)
+      } else {
+        setIsOpen((isOpen: boolean) => !isOpen)
+      }
+    }, [selection, variant, setIsOpen, setSelection])
+
+    const handleOptionSelect = useCallback(
+      (option: FilterButtonOption) => {
+        setSelection(option)
+        onSelect?.(option)
+      },
+      [onSelect]
+    )
+
+    const anchorRef = useRef<HTMLButtonElement>(null)
+
+    return (
+      <BaseButton
+        ref={ref || anchorRef}
```

Comment:
we interested in merging refs?

Reply:
not sure I know what you mean. The point of this line was just to use a ref if the user passed it in versus generating one for the sake of popup

Reply:
gotcha, i think thats probably fine then, but usually when there is an external ref and internal ref, you'd do https://github.com/AudiusProject/audius-protocol/blob/6947471f19b90e1f881496388f8074690a21afd4/packages/web/src/components/data-entry/TextAreaV2.tsx#L132

Reply:
in this case not really needed cause you dont calculate things with the inner ref, but a headsup

Reply:
yeah gotcha, i dont think we want to merge. it can just be one ref